src/IncludedCommands/UsefulFun/fly.lua

```diff
@@ -247,42 +267,101 @@ return {
+                        local X,Y = nil,nil
+                        if Key.KeyCode == Enum.KeyCode.Thumbstick1 then
+                            X = Key.Position.X
+                            Y = Key.Position.Y
+                        elseif Key.UserInputType == Enum.UserInputType.Touch then
+                            local camera = workspace.CurrentCamera
+                            local movementDirection = camera.CFrame:vectorToObjectSpace(Flight.Humanoid.MoveDirection).Unit
+                            X = movementDirection.X
+                            Y = movementDirection.Z * -1
+                        end
+
+                        if X and Y then
+                            if X <= -.5 then
+                                Flight.KeysDown[Enum.KeyCode.A] = true
+                                Flight.KeysDown[Enum.KeyCode.D] = false
+                            elseif X >= .5 then
+                                Flight.KeysDown[Enum.KeyCode.A] = false
+                                Flight.KeysDown[Enum.KeyCode.D] = true
+                            else
+                                Flight.KeysDown[Enum.KeyCode.A] = false
+                                Flight.KeysDown[Enum.KeyCode.D] = false
+                            end
+
+                            if Y <= -.5 then
+                                Flight.KeysDown[Enum.KeyCode.S] = true
+                                Flight.KeysDown[Enum.KeyCode.W] = false
+                            elseif Y >= .5 then
+                                Flight.KeysDown[Enum.KeyCode.S] = false
+                                Flight.KeysDown[Enum.KeyCode.W] = true
+                            else
+                                Flight.KeysDown[Enum.KeyCode.S] = false
+                                Flight.KeysDown[Enum.KeyCode.W] = false
+                            end
+                        end
+
+                        Flight:UpdateMultipliers()
+                    end))
                     table.insert(Flight.Events, UserInputService.InputEnded:Connect(function(Key)
-                        if Flight.KeysDown[Key.KeyCode] ~= nil then
+                        if Flight.MultiPlatform[Key.KeyCode] ~= nil then
                             --Update the key press.
-                            Flight.KeysDown[Key.KeyCode] = false
+                            Flight.KeysDown[Flight.MultiPlatform[Key.KeyCode]] = false
                             Flight:UpdateMultipliers()
                         end
                     end))
-    
+
+                    table.insert(Flight.Events,Flight.Humanoid:GetPropertyChangedSignal("MoveDirection"):Connect(function()
+                        -- Stop the flight movement when the Humanoid has no Movement Direction
+                        local MoveDirection = Flight.Humanoid.MoveDirection
+                        if MoveDirection.X < .5 and MoveDirection.X > -.5 and MoveDirection.Z < .5 and MoveDirection.Z > -.5 then
```

Comment:
Again, leading zeros for decimals.
```suggestion
                        if MoveDirection.X < 0.5 and MoveDirection.X > -0.5 and MoveDirection.Z < 0.5 and MoveDirection.Z > -0.5 then
```